src/index/TextIndexReadWrite.cpp

```diff
@@ -7,8 +7,9 @@
 namespace textIndexReadWrite {
 
 // ____________________________________________________________________________
-void zstdCompressAndWrite(const void* src, size_t numBytes,
-                          ad_utility::File& out, off_t& currentOffset) {
+template <typename T>
+void zstdCompressAndWrite(const T* src, size_t numBytes, ad_utility::File& out,
```

Comment:
I think your current usage of this function also allows the better interface
`compressAndWrite(std::span<const T>, out, currentOffset)` , 
then you can do the `size() * sizeof(T)` inside this lowlevel function, and the call site becomes much simpler.